File: C program/Circular queue/CircularQueue.c
#include <stdio.h>
#define max 5
int queue[max];
int front = -1, rear = -1, item;

void enqueue()
{
    if (front == -1 && rear == -1)
    {
        front = rear = 0;
        queue[rear] = item;
    }
    else if ((rear + 1) % max == front)
    {
        printf("\nThe queue is full !");
    }
    else
    {
        rear = (rear + 1) % max;
    }
}

void dequeue()
{
    if (front == -1 && rear == -1)
    {
        printf("\nThe queue is empty");
    }
    else if (front == rear)
    {
        front = rear = -1;
    }
    else
    {
        item = queue[front];
        front = (front + 1) % max;
        printf("\ndeleted item is %d", item);
    }
}

void display()
{
    int i=front;
    if (front == -1 && rear == -1)
    {
        printf("\nThe queue is empty");
    }
    else
    {
        printf("\n Elements in a queue are: ");
        while (i != rear)
        {
            printf("\n queue[%d]=%d", i, queue[i]);
            i = (i + 1) % max;
        }
        printf("\n queue[%d]=%d", i, queue[rear]);
    }
}
void main()
{
    int choice = 1, x;
    while (choice < 4 && choice != 0)
    {
        printf("\n Press 1: Insert and Element");
        printf("\n Press 2: Delete an Element");
        printf("\n Press 3: Display the element");
        printf("\n Enter your choice: ");
        scanf("%d", &choice);
        switch (choice)
        {
        case 1:
            printf("enter the element which is to be inserted:");
            scanf("%d", &x);
            enqueue(x);
            break;
        case 2:
            dequeue();
            break;
        case 3:
            display();
        }
    }
}

File: QueueWithJava/Circular queue/CircularQueue.java
import java.util.Scanner;

public class CircularQueue {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Queue q = new Queue(5);
        int ch;

        do {
            System.out.println("1)Enqueue 2)Dequeue 3)Display 0)Exist");
            System.out.println("Enter your choice: ");
            ch = sc.nextInt();
            switch (ch) {
                case 1:
                    System.out.println("Enter the item:");
                    int item = sc.nextInt();
                    q.enqueue(item);
                    break;

                case 2:
                    q.dequeue();
                    break;

                case 3:
                    q.display();
                    break;

                default:
                    System.out.println("Invalid choice !!");
                    break;
            }
        } while (ch != 0);

        sc.close();
    }
}

class Queue {
    private int rear, front;
    private int max;
    private int[] queue;

    public Queue(int max) {
        this.rear = this.front = -1;
        this.max = max;
        this.queue = new int[max];
    }

    public void enqueue(int item) {

        if (front == -1 && rear == -1) {
            front = rear = 0;
            queue[rear] = item;
        } else if ((rear + 1) % max == front) {
            System.out.println("The queue is full");
        } else {
            rear = (rear + 1) % max;
            queue[rear] = item;

        }
    }

    public void dequeue() {
        int item;
        if (front == -1 && rear == -1) {
            System.out.println("The queue is empty");
        } else {
            item = queue[front];
            if (front == rear) {
                front = rear = -1;
            } else {
                front = (front + 1) % max;
                System.out.println("Deleted item is " + item);
            }
        }
    }

    public void display() {
        if (front == -1 && rear == -1) {
            System.out.println("The queue is empty");
        } else {
            int i = front;
            System.out.print("Queue: ");
            while (i != rear) {
                System.out.print(queue[i] + " ");
                i = (i + 1) % max;
            }
            System.out.print(queue[i] + " ");
            System.out.println();
        }
    }
}